airflow/providers/apache/spark/decorators/pyspark_submit.py

```diff
@@ -0,0 +1,205 @@
+# under the License.
+from __future__ import annotations
+
+import inspect
+import os
+import pickle
+import warnings
+from tempfile import TemporaryDirectory
+from textwrap import dedent
+from typing import TYPE_CHECKING, Callable, Sequence
+
+import dill
+
+from airflow.configuration import conf
+from airflow.decorators.base import DecoratedOperator, TaskDecorator, task_decorator_factory
+from airflow.exceptions import AirflowException
+from airflow.providers.apache.spark.decorators.pyspark import SPARK_CONTEXT_KEYS
+from airflow.providers.apache.spark.operators.spark_submit import SparkSubmitOperator
+from airflow.utils.python_virtualenv import write_python_script
+
+if TYPE_CHECKING:
+    from airflow.utils.context import Context
+
+
+class _PysparkSubmitDecoratedOperator(DecoratedOperator, SparkSubmitOperator):
+    custom_operator_name = "@task.pyspark_submit"
+
+    template_fields: Sequence[str] = (
+        "conf",
+        "files",
+        "py_files",
+        "jars",
+        "driver_class_path",
+        "packages",
+        "exclude_packages",
+        "keytab",
+        "principal",
+        "proxy_user",
+        "name",
+        "env_vars",
+        "properties_file",
+        "op_args",
+        "op_kwargs",
+    )
+
+    def __init__(
+        self,
+        python_callable: Callable,
+        op_args: Sequence | None = None,
+        op_kwargs: dict | None = None,
+        use_dill: bool = False,
+        expect_airflow: bool = False,
+        **kwargs,
+    ):
+        self.use_dill = use_dill
+        self.expect_airflow = expect_airflow
+
+        signature = inspect.signature(python_callable)
+        parameters = [
+            param.replace(default=None) if param.name in SPARK_CONTEXT_KEYS else param
+            for param in signature.parameters.values()
+        ]
+        # mypy does not understand __signature__ attribute
+        # see https://github.com/python/mypy/issues/12472
+        python_callable.__signature__ = signature.replace(parameters=parameters)  # type: ignore[attr-defined]
+
+        if kwargs.get("application"):
+            if not conf.getboolean("operators", "ALLOW_ILLEGAL_ARGUMENTS"):
+                raise AirflowException(
+                    "Invalid argument 'application' were passed to `@task.pyspark_submit`."
+                )
+            warnings.warn(
+                "Invalid argument 'application' were passed to @task.pyspark_submit.",
+                UserWarning,
+                stacklevel=2,
+            )
+        if kwargs.get("application_args"):
+            if not conf.getboolean("operators", "ALLOW_ILLEGAL_ARGUMENTS"):
+                raise AirflowException(
+                    "Invalid argument 'application_args' were passed to `@task.pyspark_submit`."
+                )
+            warnings.warn(
+                "Invalid argument 'application_args' were passed to `@task.pyspark_submit`.",
+                UserWarning,
+                stacklevel=2,
+            )
```

Comment:
I wanted to include it in the kwargs-related exceptions of `BaseOperator`, 
but I haven't yet found a way to exclude it from the reserved kwargs of the `SparkSubmitOperator`.
Any ideas?

https://github.com/apache/airflow/blob/e827bfbd2c79f3432411e14351c245a06410766a/airflow/models/baseoperator.py#L931-L944